acracensor/handlers/querycapture_handler.go

```diff
@@ -5,89 +5,127 @@ import (
 	"io/ioutil"
 	"encoding/json"
 	"os"
+	"github.com/cossacklabs/acra/logging"
+	log "github.com/sirupsen/logrus"
+	"time"
+	"os/signal"
+	"syscall"
 )
 
+const MaxQueriesInChannel = 10
+const TimeoutSecondsToSerialize = 1
+
 type QueryCaptureHandler struct {
 	Queries []QueryInfo
 	filePath string
-	occuredError *CaptureError
-
 	logChannel chan QueryInfo
+	signalToSerialize chan bool
 }
 
 type QueryInfo struct {
 	RawQuery string
 	IsForbidden bool
 }
 
-type CaptureError struct {
-	err error
-	query *QueryInfo
-}
-
 func NewQueryCaptureHandler(filePath string) (*QueryCaptureHandler, error) {
-	file, err := os.OpenFile(filePath, os.O_RDONLY|os.O_CREATE, 0600)
-	if err != nil {
-		return nil, err
+
+	var _, err = os.Stat(filePath)
+
+	// create file if not exists
+	if os.IsNotExist(err) {
+		var file, err = os.Create(filePath)
+		if err != nil {
+			return nil, err
+		}
+		defer file.Close()
 	}
-	err = file.Close()
+
+	bufferBytes, err := ioutil.ReadFile(filePath)
 	if err != nil {
 		return nil, err
 	}
 
-	emptyQuery := &QueryInfo{RawQuery:"", IsForbidden:false}
-	occuredError := &CaptureError{err:nil, query:emptyQuery}
+	var queries []QueryInfo
 
-	logChannel_ := make(chan QueryInfo)
+	if len(bufferBytes) != 0 {
+		if err = json.Unmarshal(bufferBytes, &queries); err != nil{
+			return nil, err
+		}
+	}
 
+	logChannel := make(chan QueryInfo, MaxQueriesInChannel)
+	signalToSerialize := make(chan bool)
+	signalShutdown := make(chan os.Signal, 2)
+	signal.Notify(signalShutdown, os.Interrupt, syscall.SIGTERM)
+
+	//serialization signal
+	go func() {
+		for {
+			time.Sleep(TimeoutSecondsToSerialize * time.Second)
```

Comment:
If I add timeout parameter to constructor, It will be necessary to add this parameter to yaml config. Is it good? May be it is ok to initialise it with default value (1 second) and give possibility to change via getter/setter?